ast/parser.go

```diff
@@ -349,6 +353,7 @@ func (p *Parser) parseRules() []*Rule {
 
 		next.Head = rule.Head.Copy()
 		setLocRecursive(next.Head, loc)
+		loc.Text = p.s.Text(loc.Offset, p.s.lastEnd)
```

Comment:
Nah, just needed to be somewhere in that group of statements after `parseBody()` and before `p.scan()`.

I'm happy to re-arrange and put it before we do `setLocRecursive()`

Reply:
As i'm staring at this i'm actually having second thoughts on whether this is correct. The `next.Head` probably should have the same text and location as the original rule's head (which it copied). I'm not sure it makes sense for it to have the location of the entire rule.

eg
```
p {
  input.x > 1  
} {
  input.y < 0
}
```

The second rule is parsed into  
```
p {
  input.y < 0
}
```
So it would make sense that the location for the head points back to the original `p` location versus having the text and location:

```
{
  input.y < 0
}
```
(which is what the rule's location text is)

Reply:
> The next.Head probably should have the same text and location as the original rule's head (which it copied).

These chained rules are kind of an anti-pattern IMO.

Making the 2nd rule head location be equal to the 1st rule head location could be annoying if there's a safety error. For example:

```
p = x {
   x = 1
} {
   true
}
```

In this case, the 2nd rule is unsafe because 'x' is implicitly contained in the 2nd rule head but not assigned in the body. If the 2nd rule head and the terms contained inside it have their location set to the 1st rule head location, the safety error could report a line number that's not close to the source of the problem (it'll depend on what location is used...)